pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/GrpcReceivingMailbox.java

```diff
@@ -85,4 +97,28 @@ private boolean waitForInitialize()
   public String getMailboxId() {
     return _mailboxId;
   }
+
+  /**
+   * Converts the data sent by a {@link GrpcSendingMailbox} to a {@link TransferableBlock}.
+   *
+   * @param mailboxContent data sent by a GrpcSendingMailbox.
+   * @return null if the mailboxContent passed is null or empty. Will return an error block if the returned DataBlock
+   *         contains exceptions.
+   * @throws IOException if the MailboxContent cannot be converted to a TransferableBlock.
+   */
+  private TransferableBlock fromMailboxContent(@Nullable MailboxContent mailboxContent)
+      throws IOException {
+    if (mailboxContent == null) {
+      return null;
+    }
+    ByteBuffer byteBuffer = mailboxContent.getPayload().asReadOnlyByteBuffer();
+    if (byteBuffer.hasRemaining()) {
+      BaseDataBlock dataBlock = DataBlockUtils.getDataBlock(byteBuffer);
+      if (dataBlock instanceof MetadataBlock && !dataBlock.getExceptions().isEmpty()) {
+        return TransferableBlockUtils.getErrorTransferableBlock(dataBlock.getExceptions());
+      }
+      return new TransferableBlock(dataBlock);
+    }
+    return null;
+  }
```

Comment:
tag this as `@Nullable` return.
on a second thought, should we ever return a null block?